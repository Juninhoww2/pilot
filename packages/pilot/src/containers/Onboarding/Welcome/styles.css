@import "former-kit-skin-pagarme/dist/styles/webfonts/poppins/index.css";
@import "former-kit-skin-pagarme/dist/styles/colors.css";
@import "former-kit-skin-pagarme/dist/styles/spacing.css";

.onboardingWelcome {
  display: flex;
  justify-content: center;
  flex-direction: column;
  height: 100%;
}

.welcome {
  margin: 0;
  font-size: 14px;
  line-height: 20px;
}

.welcomeText {
  font-size: 14px;
  line-height: 20px;
  margin: 0 0 var(--spacing-tiny) 0;
}

.welcomeDescription {
  font-size: 14px;
  line-height: 20px;
  margin: 0 0 0 48px;
}

.welcomeTitle {
  color: var(--color-birdperson-gray-600);
  font-family: "Poppins", sans-serif;
  font-size: 32px;
  font-weight: 600;
  line-height: 46px;
  margin-bottom: var(--spacing-tiny);
}

.welcomeDescriptionTitle {
  display: flex;
  align-items: center;
  margin-top: var(--spacing-medium);

  & p {
    margin: 0 0 0 var(--spacing-small);
    font-weight: bold;
  }

  & svg {
    color: var(--color-barney-purple-100);
  }
}

.welcomeContent {
  margin-bottom: 42px;
}
