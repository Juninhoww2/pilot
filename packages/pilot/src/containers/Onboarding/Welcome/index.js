import React from 'react'
import PropTypes from 'prop-types'
import { Button } from 'former-kit'
import HeartRate from 'emblematic-icons/svg/HeartRate32.svg'
import Shield from 'emblematic-icons/svg/Shield32.svg'
import Target from 'emblematic-icons/svg/Target32.svg'

import styles from './styles.css'

const Welcome = ({ handleStartOnboarding, t, userName }) => (
  <div className={styles.onboardingWelcome}>
    <p className={styles.welcome}>{t('pages.onboarding.welcome.header', { userName })}</p>
    <h3 className={styles.welcomeTitle}>{t('pages.onboarding.welcome.title')}</h3>
    <p className={styles.welcomeText}>{t('pages.onboarding.welcome.description')}</p>
    <div className={styles.welcomeContent}>
      <div className={styles.welcomeDescriptionTitle}>
        <Target />
        <p>{t('pages.onboarding.welcome.documentation')}</p>
      </div>
      <p className={styles.welcomeDescription}>{t('pages.onboarding.welcome.documentation_description')}</p>
      <div className={styles.welcomeDescriptionTitle}>
        <Shield />
        <p>{t('pages.onboarding.welcome.antifraud')}</p>
      </div>
      <p className={styles.welcomeDescription}>{t('pages.onboarding.welcome.antifraud_description')}</p>
      <div className={styles.welcomeDescriptionTitle}>
        <HeartRate />
        <p>{t('pages.onboarding.welcome.monitoring')}</p>
      </div>
      <p className={styles.welcomeDescription}>{t('pages.onboarding.welcome.monitoring_description')}</p>
    </div>
    <Button
      onClick={handleStartOnboarding}
      size="huge"
      fullWidth
    >
      {t('pages.onboarding.welcome.advance')}
    </Button>
  </div>
)

Welcome.propTypes = {
  handleStartOnboarding: PropTypes.func,
  t: PropTypes.func.isRequired,
  userName: PropTypes.string.isRequired,
}

Welcome.defaultProps = {
  handleStartOnboarding: () => {},
}

export default Welcome
