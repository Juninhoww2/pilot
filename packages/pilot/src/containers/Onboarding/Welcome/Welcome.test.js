import React from 'react'
import { render } from '@testing-library/react'
import Welcome from '.'

test('Welcome should render', () => {
  const { container } = render(
    <Welcome
      t={value => value}
      userName="Name"
    />
  )

  expect(container).toMatchSnapshot()
})
