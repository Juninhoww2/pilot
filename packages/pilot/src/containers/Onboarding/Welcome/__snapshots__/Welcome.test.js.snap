// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render 1`] = `
<div>
  <div
    class="onboardingWelcome"
  >
    <p
      class="welcome"
    >
      pages.onboarding.welcome.header
    </p>
    <h3
      class="welcomeTitle"
    >
      pages.onboarding.welcome.title
    </h3>
    <p
      class="welcomeText"
    >
      pages.onboarding.welcome.description
    </p>
    <div
      class="welcomeContent"
    >
      <div
        class="welcomeDescriptionTitle"
      >
        <svg
          data-file-name="SvgTarget32"
        />
        <p>
          pages.onboarding.welcome.documentation
        </p>
      </div>
      <p
        class="welcomeDescription"
      >
        pages.onboarding.welcome.documentation_description
      </p>
      <div
        class="welcomeDescriptionTitle"
      >
        <svg
          data-file-name="SvgShield32"
        />
        <p>
          pages.onboarding.welcome.antifraud
        </p>
      </div>
      <p
        class="welcomeDescription"
      >
        pages.onboarding.welcome.antifraud_description
      </p>
      <div
        class="welcomeDescriptionTitle"
      >
        <svg
          data-file-name="SvgHeartRate32"
        />
        <p>
          pages.onboarding.welcome.monitoring
        </p>
      </div>
      <p
        class="welcomeDescription"
      >
        pages.onboarding.welcome.monitoring_description
      </p>
    </div>
    <button
      class="undefined"
      type="button"
    >
      <span>
        pages.onboarding.welcome.advance
      </span>
      <span
        style="height: 0px; left: 0px; top: 0px; width: 0px;"
      />
    </button>
  </div>
</div>
`;
